.github/actions/release_artifacts/action.yml

```diff
@@ -0,0 +1,178 @@
+        ghcr.io/edgelesssys/contrast/node-installer-kata:latest=$nodeInstallerKataImgTagged
+        ghcr.io/edgelesssys/contrast/node-installer-kata-gpu:latest=$nodeInstallerKataGPUImgTagged
+        ghcr.io/edgelesssys/contrast/tardev-snapshotter:latest=$tardevSnapshotterImgTagged
+        EOF
+    - name: Upload image replacement file (for main branch PR)
+      uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+      with:
+        name: image-replacements
+        path: |
+          image-replacements.txt
+    - name: Update cli assets
+      shell: bash
+      run: |
+        cp image-replacements.txt cli/cmd/assets/image-replacements.txt
+        git add cli/cmd/assets/image-replacements.txt
+        git diff --staged --quiet || git commit -m "release: update cli assets"
+    - name: Expected artifact paths
+      id: artifact-paths
+      shell: bash
+      run: |
+        cat << 'EOF' | tee -a "${GITHUB_OUTPUT}"
+        paths<<EOM
+        workspace/contrast-cli/bin/contrast
+        workspace/coordinator*.yml
+        workspace/runtime-*.yml
+        workspace/emojivoto-demo.yml
+        workspace/mysql-demo.yml
+        workspace/vault-demo.yml
+        EOM
+        EOF
+    - name: Create coordinator resource definitions
+      shell: bash
+      run: |
+        mkdir -p workspace
+
+        nix shell .#contrast --command resourcegen \
+          --image-replacements "./image-replacements.txt" \
+          --add-load-balancers \
+          coordinator > "workspace/coordinator.yml"
+    - name: Create runtime resource definitions
+      shell: bash
+      run: |
+        platforms=(
+          "aks-clh-snp"
+          "k3s-qemu-snp-gpu"
+          "k3s-qemu-snp"
+          "k3s-qemu-tdx"
+          "metal-qemu-snp-gpu"
+          "metal-qemu-snp"
+          "metal-qemu-tdx"
+          "rke2-qemu-tdx"
+        )
+        for platform in "${platforms[@]}"; do
+          nix shell .#contrast --command resourcegen \
+            --image-replacements ./image-replacements.txt \
+            --platform "$platform" \
+            runtime > "workspace/runtime-$platform.yml"
+        done
+    - name: Create demo resource definitions
+      shell: bash
+      run: |
+        nix shell .#contrast --command resourcegen \
+          --image-replacements ./image-replacements.txt \
+          --add-load-balancers emojivoto-sm-ingress > workspace/emojivoto-demo.yml
+        nix shell .#contrast --command resourcegen \
+          --image-replacements ./image-replacements.txt \
+          --add-load-balancers mysql > workspace/mysql-demo.yml
+        nix shell .#contrast --command resourcegen \
+          --image-replacements ./image-replacements.txt \
+          --add-load-balancers vault > workspace/vault-demo.yml
+    - name: Build CLI
+      shell: bash
+      run: |
+        nix build -L .#contrast-cli-release --out-link workspace/contrast
+    - name: AWS login (IAM role)
```

Comment:
Not sure what the `IAM role` is supposed to tell me - I was half expecting the role to be an IAM admin...

Reply:
Now I'm even more confused - aren't we using the OIDC integration here? 

https://github.com/aws-actions/configure-aws-credentials/blob/68858db075f07f5ade3752254dfda34afd669f9b/README.md?plain=1#L84

Reply:
Yes, this is the "Assume [IAM] Role directly using GitHub OIDC provider" flow. Just as always. I just copied that from another workflow without any change. Not sure where the confusion is.